From 7c19155d3444c05d4e8d45dbac00e033df3eafc1 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Sat, 24 Mar 2018 21:31:02 +0200
Subject: [PATCH 2/3] init: Don't run update_sys_usb_config if /data isn't
 mounted

* When /data is not mounted / not yet decrypted,
  persistent property overrides are not loaded yet
  so persist.sys.usb.config value is the one from default.prop.

  This caused adb to always be disabled on boot even if
  it was enabled in development settings.

Change-Id: Ic2a51591ad6b77f3463cfa91e9d7362410a021f2
---
 init/property_service.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 25517edbd..5043fa224 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -681,7 +681,9 @@ void property_load_boot_defaults() {
     load_properties_from_file("/odm/default.prop", NULL);
     load_properties_from_file("/vendor/default.prop", NULL);
 
-    update_sys_usb_config();
+    if (android::base::GetBoolProperty("ro.persistent_properties.ready", false)) {
+        update_sys_usb_config();
+    }
 }
 
 static void load_override_properties() {
-- 
2.17.0

